### PR TITLE
Model Validation and Unit Tests

### PR DESCRIPTION
**Pull request by MarcoDDM**
**Description**

This PR focuses on improving model validation and adding RSpec unit tests for the `Comment`, `Like`, `Post`, and `User `models.

**Changes**
- Strengthened model validation logic.
- Added comprehensive RSpec unit tests.
- Testing
- All changes are rigorously tested using RSpec to ensure accurate model behavior.
- Added screenshot for code reviewer's convenience on the testing aspect.

rspecs_test

Thanks for your time and review. Don't hesitate to contact me if there are any changes needed.

### CODE REVIEW

File: spec/models/comment_spec.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+require 'rails_helper'
+
+RSpec.describe Comment, type: :model do
+  let(:tom) { User.create(name: 'Tom', posts_counter: 0) }
+  let(:post) { Post.create(title: 'Title', comments_counter: 0, likes_counter: 0, author: tom) }
+end
```
Comment (franclobo) [inline spec/models/comment_spec.rb]: - [ ] What are you testing here? Please 🙏🏽 validate the comment when it should have a text, it is valid with an author and a post. It is a requirement for this milestone 👇🏽 

> Add unit specs for all of your models' methods and validations.

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @MarcoDDM, 

# Required Changes ♻️

so-close-jim-halpert

Good job so far!
You are doing it well 👏🏽 there are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct Git flow ✔️ 
- Use validations for models. ✔️ 
- Write unit tests for models. ✔️ 
- All tests passed ✔️ 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/comment_spec.rb (hunk 1 above)
Comment (MarcoDDM) [reply spec/models/comment_spec.rb]: I was so absorbed in that moment that I guess I skipped it. I will do that. Thank you.

**Review by mwanawabangona — APPROVED** [review]
Hi @MarcoDDM 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations!  


## Highlights
*   No linter errors :heavy_check_mark:
*   Repo looks professional :briefcase:
*   Solid pull request scroll
*   Good quality commits
*   Good job implementing the requested changes!! :clap: :clap: :clap:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**